dcp-0005/dcp-0005.mediawiki

```diff
@@ -0,0 +1,899 @@
+### Read the unary-encoded quotient <code>q</code> from the bit stream as follows:
+#### Set <code>q</code> to 0
+#### Read from the bit stream until a bit with a value of 0 is encountered while incrementing <code>q</code> by 1 for every bit read excluding the final 0
+### Read the remainder <code>r</code> from the bit stream as a big-endian with <code>B</code> bits
+### Calculate the difference <code>d</code> as <code>d = (q << B) + r</code>
+## Reconstruct the hash value that represents the original set item by incrementing the intermediate sum by the decoded difference <code>d</code> (<code>s += d</code>)
+## Return true if the search term is equal to the intermediate sum <code>s</code>
+## Return false if intermediate sum <code>s</code> is greater than the search term
+# Return false since there are no longer any items in the GCS to search
+
+===Version 2 Block Filters===
+
+Version 2 block filters MUST be a [[#golomb-coded-sets|GCS]] loaded with the
+items from the block as described in the [[#Contents|Contents]] section and
+constructed with the following parameters:
+
+* The bit size parameter <code>B</code> set to 19
+* The false positive rate parameter <code>M</code> set to 784931 <sup>[ [[#optimal-gcs-parameters|why?]] ]</sup>
+* The <code>key</code> set to the first 16 bytes of the <code>MerkleRoot</code> (in the exact order produced by the hash function) of the block covered by the filter
+
+====Contents====
+
+The following special cases MUST be applied to all items in both the regular and
+stake transaction trees:
+
+* Scripts that are not version 0, empty, or larger than the max allowed script size, <code>16384</code> bytes, MUST NOT be included
+* Output scripts for transactions in the stake tree MUST NOT include the initial stake opcode tag (OP_SS*)
+** NOTE: This applies both to when they are created and when they are redeemed by transaction inputs
+
+While considering the aforementioned exceptions, the following items MUST be
+included:
+
+* For transactions in the regular transaction tree:
+** Previous output scripts referenced by the transaction inputs, with the exception of the coinbase which has no inputs
+** All output scripts of the transaction outputs
+* For transactions in the stake transaction tree:
+** For ticket purchases:
+*** Previous output scripts referenced by the transaction inputs
+*** Commitment output scripts converted from the commitment hash and type to the full script
+*** Change output scripts
+** For votes:
+*** Subsidy generation output scripts
+*** Output scripts that pay the original ticket commitments
+** For revocations:
+*** Output scripts that pay the original ticket commitments
+
+Note that the voting rights output of ticket purchases and references to it by
+votes and revocations is intentionally NOT included because lightweight clients,
+by definition, do not have access to the full blocks, so they can't reasonably
+vote on their validity.
```

Comment:
Although I harped on this sentence in Matrix, I don't have any specific changes that I would requests.  The reason I questioned this is because unless it is absolutely necessary to prevent this, I'd be concerned that omitting the stakesubmission output/prevout scripts from the contents of the GCS could create unforeseen limitations.  However, since the sstxcommitment and sstxchange output data is included, the lightweight clients should not lack any info.

Reply:
I can add the following if you think it would help clarify:

> Also, lightweight clients are able to discover the relevant transactions via the their reward commitments and change outputs.

Reply:
That's a good addition, I think.